packages/core/src/components/axes/axis.ts

```diff
@@ -65,6 +66,44 @@ export class Axis extends Component {
 		this.services.events.dispatchEvent(Events.Model.UPDATE);
 	};
 
+	// decide if the tick rotation is required
+	isTickRotationRequired(rotateTicksBySpaceCalculation: boolean) {
+		// if zoomDomain is changing
+		if (this.zoomDomainChanging) {
+			const { position: axisPosition } = this.configs;
+			const axisOptions = Tools.getProperty(
+				this.model.getOptions(),
+				"axes",
+				axisPosition
+			);
+			// user could decide if tick rotation is required during zoom domain changing
+			const rotateWhileZooming = Tools.getProperty(
+				axisOptions,
+				"ticks",
+				"rotateWhileZooming"
+			);
+
+			if (
+				!rotateWhileZooming || // default to always
+				rotateWhileZooming === TickRotations.ALWAYS
+			) {
+				// if option is set to ALWAYS or not set
+				return true;
+			} else if (rotateWhileZooming === TickRotations.NEVER) {
+				// if option is set to NEVER
+				return false;
+			} else if (rotateWhileZooming === TickRotations.DEPENDING) {
```

Comment:
this should probably be called `AUTO`

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/770/commits/b08826d54b5d951caef0a37a018920ebb2acfca9.